### ISSUE TITLE
Wrong bits getting written/read in BigEndian mode?

### ISSUE THREAD

**Issue by cswinter** [icewind1991/bitbuffer#9]
I seem to have run into some edge case where bitbuffer ends up emitting (and reading) the incorrect sequence of bits.
Minimal repro in https://github.com/cswinter/bitwriter-minimal-repro/blob/main/src/main.rs:

```rust
fn main() {
    let mut bytes = vec![];
    let mut writer = bitbuffer::BitWriteStream::new(&mut bytes, bitbuffer::BigEndian);
    let num1 = 0b11000_00111110u64;
    let num2 = 0b1111111_11100100_00100100_11011101_00000011_11100000_01100111_11011011u64;
    writer.write_int(num1, 13).unwrap();
    writer.write_int(num2, 63).unwrap();

    let buffer = bitbuffer::BitReadBuffer::new(&bytes, bitbuffer::BigEndian);
    let mut reader = bitbuffer::BitReadStream::new(buffer);
    let num1actual = reader.read_int::<u64>(13).unwrap();
    let num2actual = reader.read_int::<u64>(63).unwrap();

    println!("EXPECTED: {:013b}{:064b}", num1, num2);
    print!("BYTES:    ");
    for byte in bytes { print!("{:08b}", byte); }
    println!();
    println!("READ:     {:013b}{:064b}", num1actual, num2actual);
    assert_eq!(num1actual, num1);
    assert_eq!(num2actual, num2);
}
```

This gives me the following output:

```
(base) clemens@LAPTOP-NV53OUEJ:~/src/bitwriter-minimal-repro$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/bitwriter-minimal-repro`
EXPECTED: 11000001111100111111111100100001001001101110100000011111000000110011111011011
BYTES:    11000001111100111111111100100001001001101110100000011110000001100111110110110000
READ:     11000001111100011111111110010000100100110111010000001111000000110011111011011
thread 'main' panicked at src/main.rs:20:5:
assertion `left == right` failed
  left: 4607765634588108763
 right: 9215531269178288091
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The 63 bit write seems to randomly delete a 1 bit in the middle of the u64, and then when reading it back out a `0` seems to be added at the beginning. The same code runs without triggering the assertion when using `LittleEndian` mode.

**Comment by hussein-aitlahcen** [comment]
Can you check again on latest master to see if https://github.com/icewind1991/bitbuffer/pull/6 fixed it?

**Comment by icewind1991** [comment]
That doesn't seem to fix it

**Comment by hussein-aitlahcen** [comment]
> That doesn't seem to fix it

Oops deleted because it was not making any sense haha!

**Comment by icewind1991** [comment]
Should be fixed with b46bc881f0922c376e69b4eaee4efbd3b77f409b, released with 0.11
